src/Infrastructure/LeanCode.Kratos/KratosAuthenticationHandler.cs

```diff
@@ -76,49 +76,49 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         }
     }
 
-    protected virtual Task<KratosSession?> GetSessionAsync()
+    protected virtual async Task<IToSessionApiResponse?> GetSessionAsync()
     {
         if (Request.Headers.TryGetValue("X-Session-Token", out var token) && !string.IsNullOrEmpty(token))
         {
-            return api.ToSessionAsync(xSessionToken: token);
+            return await api.ToSessionAsync(xSessionToken: token.ToString());
         }
         else if (
             AuthenticationHeaderValue.TryParse(Request.Headers.Authorization, out var value)
             && string.Equals(value.Scheme, "bearer", StringComparison.OrdinalIgnoreCase)
             && !string.IsNullOrEmpty(value.Parameter)
         )
         {
-            return api.ToSessionAsync(xSessionToken: value.Parameter);
+            return await api.ToSessionAsync(xSessionToken: value.Parameter);
         }
         else if (
             Request.Cookies.TryGetValue(Options.SessionCookieName, out var cookie) && !string.IsNullOrEmpty(cookie)
         )
         {
-            return api.ToSessionAsync(cookie: $"{Options.SessionCookieName}={cookie}");
+            return await api.ToSessionAsync(cookie: $"{Options.SessionCookieName}={cookie}");
         }
         else
         {
-            return Task.FromResult(null as KratosSession);
+            return null;
         }
     }
 
     protected virtual List<Claim> ExtractClaims(KratosSession session)
     {
         var claims = new List<Claim>
         {
-            new(Options.NameClaimType, session.Identity.Id),
-            new("iss", api.Configuration.BasePath),
-            new("iat", ToUnixTimeSecondsString(session.IssuedAt)),
-            new("exp", ToUnixTimeSecondsString(session.ExpiresAt)),
-            new("auth_time", ToUnixTimeSecondsString(session.AuthenticatedAt)),
+            new(Options.NameClaimType, session.Identity!.Id),
+            new("iss", SchemaUrlPathRegex().Replace(session.Identity.SchemaUrl, string.Empty)),
```

Comment:
We're now chopping off the schema URL path from the response, which should result in the base URL under which Kratos is publically accessible.